README.md

```diff
@@ -55,8 +69,222 @@ documentation for more details.
+
+```ts
+import { App } from 'aws-cdk-lib';
+import { ShellStep } from 'aws-cdk-lib/pipelines';
+import { GithubWorkflow } from 'cdk-pipelines-github';
+
+const app = new App();
+
+const pipeline = new GithubWorkflow(app, 'Pipeline', {
+  synth: new ShellStep('Build', {
+    commands: [
+      'yarn install',
+      'yarn build',
+    ],
+  }),
+  awsCredentials: {
+    accessKeyId: 'MY_ID',
+    secretAccessKey: 'MY_KEY',
+    sessionToken: 'MY_TOKEN', // default is no token
+  },
+});
+```
+
+### OpenId Connect
+
+Authenticating via OpenId Connect means you do not need to store long-lived 
+credentials as GitHub Secrets. With OIDC, you provide a pre-provisioned IAM
+role to your GitHub Workflow via the `awsOidcRoleArn` property.
+
+```ts
+import { App } from 'aws-cdk-lib';
+import { ShellStep } from 'aws-cdk-lib/pipelines';
+import { GithubWorkflow } from 'cdk-pipelines-github';
+
+const app = new App();
+
+const pipeline = new GithubWorkflow(app, 'Pipeline', {
+  synth: new ShellStep('Build', {
+    commands: [
+      'yarn install',
+      'yarn build',
+    ],
+  }),
+  awsOidcRoleArn: 'arn:aws:iam::<account-id>:role/GithubActionRole',
+});
+```
+
+You can create this role either by manually setting it up, or taking advantage
+of the `AwsOidc` construct (recommended).
+
+#### `AwsOidc` Construct
+
+Because this construct involves creating an IAM role in your account, it must
+be created separate to your GitHub Workflow and deployed via a normal
+`cdk deploy` with your local AWS credentials. Upon successful deployment, the
+arn of your newly created IAM role will be exposed as a `CfnOutput`.
+
+To utilize this construct, create a separate CDK stack with the following code
+and `cdk deploy`:
+
+```ts
+import { AwsOidc } from 'cdk-pipelines-github';
+import { App, Construct, Stack, StackProps } from 'aws-cdk-lib';
+
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+    };
+  }
+}
+
+const app = new App();
+new MyAwsOidcRole(app, 'MyAwsOidcRole');
+app.synth();
+```
+
+Note: If you have previously created the GitHub identity provider with url
+`https://token.actions.githubusercontent.com`, the above example will fail
+because you can only have one such provider defined per account. In this
+case, you must provide the already created provider into your `AwsOidc`
+construct via the `provider` property.
+
+```ts
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+      provider: iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(
+        this,
+        'github',
+        'arn:aws:iam::<account-id>:oidc-provider/token.actions.githubusercontent.com',
```

Comment:
If this is always the same: you can make a static function on the class to retrieve it. Use `Aws.ACCOUNT_ID` to make it use the "current account". We can then write this:

```ts
 const provider = new AwsOidc(this, 'oidc-role', {
      repoString: 'myUser/myRepo',
      provider: AwsOidc.existingGitHubActionsProvider(),
  });
```

Pretty nice.